spyder/plugins/outlineexplorer/widgets.py

```diff
@@ -722,7 +722,7 @@ def __sort_toplevel_items(self):
                 self.ordered_editor_ids if
                 self.editor_items.get(e_id) is not None]
 
-        if current_ordered_items != new_ordered_items:
+        if current_ordered_items is not new_ordered_items:
```

Comment:
We can have this previous one, as `is not` matches for the memory location whereas `!=` looks for values. Sorry if I am wrong.
```suggestion
        if current_ordered_items != new_ordered_items:
```

Reply:
You are right, strictly speaking the patch is not correct. I haven’t noticed this because it should be safe to remove the `if` clause, i.e. it is safe to always execute the code inside the `if` block (`is not` always returns `True` here because two distinct `list` object are compared). The `if` clause looks like some kind of optimization, which only applies in the following case (as far as I can tell):
* The "Show all files" option is enabled in the Outline plugin
* The tabs in the editor are alphabetically ordered and the "Sort files alphabetically" option is _off_
* The "Sort files alphabetically" option is toggled.

There are two options:
* Remove the `if` clause
* Add compatibility code for PySide2 (which could differentiate if the version is  `<= 5.15.0`. The bug which fixes `==`/`!=` has been fixed in `5.15.1` [1])

[1] https://bugreports.qt.io/browse/PYSIDE-74
[2] https://codereview.qt-project.org/c/pyside/pyside-setup/+/312945

Reply:
> The bug which fixes ==/!= has been fixed in 5.15.1

Well, since the support for PySide2 is brand new, you could require `>=5.15.1` and leave the `if` in place.

Reply:
Do you mean the global requirement (the one checked in `spyder/requirements.py` at startup) or should the following code be used in `outlineexplorer/widgets.py`:
```py
        # Always update if PySide <= 5.15.0 is used because == and != comparisons are not supported
        # for the data types inside the lists
        update = (
            (PYSIDE2 and parse_version(PYSIDE_VERSION) <= parse_version("5.15.0"))
            or (current_ordered_items != new_ordered_items)
        )
        if update:
            …
```

I prefer the latter or completely removing the `if` clause, but the decision is up to you.

Reply:
Ok, let's go with the code you posted above then. I'm fine with it.

Reply:
Done in force push on 2021-09-20